tests/recovery.py

```diff
@@ -159,26 +238,25 @@ def run(args):
     ) as network:
         network.start_and_join(args)
 
-        for i in range(args.recovery):
+        for i in range(recoveries_count):
             # Issue transactions which will required historical ledger queries recovery
             # when the network is shutdown
             network.txs.issue(network, number_txs=1)
             network.txs.issue(network, number_txs=1, repeat=True)
 
             # Alternate between recovery with primary change and stable primary-ship,
             # with and without snapshots
-            if i % 2 == 0:
+            if i % recoveries_count == 0:
                 if args.consensus != "BFT":
-                    recovered_network = test_share_resilience(
-                        network, args, from_snapshot=True
-                    )
-                else:
-                    recovered_network = network
+                    network = test_share_resilience(network, args, from_snapshot=True)
```

Comment:
AFAICT the BFT branch isn't actually executed in our tests, is it?

Reply:
I don't think it currently is for this test. We have this check sprinkled in a few places and should probably clean it up at some point but I'll leave that to a separate PR.